packages/fs-detectors/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -2281,6 +2281,30 @@ describe('Test `detectBuilders` with `featHandleMiss=true`', () => {
       },
     ]);
   });
+
+  it('should not add middleware builder when building "nextjs"', async () => {
+    const files = ['package.json', 'pages/index.ts', 'middleware.ts'];
+    const pkg = {
+      scripts: { build: 'next build' },
+      dependencies: { next: '12.2.0' },
+    };
+    const projectSettings = {
+      framework: null,
```

Comment:
Lets add `createdAt` and set an old date since I think that is the only time we should be detecting Next.js from the dependencies (I believe this was to grandfather in old projects when we introduced project settings)

Reply:
I wrote this fix not for old projects, but because framework detection was not working in the one-click deploy button. The problem has already been fixed so this PR is meaningful only for:

- Apps incorrectly configured by users (settings "others" for Next.js).
- When framework detection become broken again for some reason in the future.

Both will unlikely happen but if happens they would cause relatively hard-to-debug behavior: build succeeds, but only middleware might behave differently.

Reply:
If its not for old projects, then lets close it because thats the only valid scenario for framework "Other" doing any framework detection.

Reply:
At first I wrote this PR for fixing the one-click deploy button the problem but indeed it's still helpful for old projects. Will update this for old projects 👍